pkg/repository/switch.go

```diff
@@ -54,37 +49,34 @@ func (r *switchRepository) Register(ctx context.Context, req *infrav2.SwitchServ
 		if err != nil {
 			return nil, err
 		}
+
 		return r.replace(ctx, old, new)
 	}
 
-	updateReq := &adminv2.SwitchServiceUpdateRequest{
-		Id:             new.Id,
+	return r.s.Switch().Update(ctx, new.Id, &adminv2.SwitchServiceUpdateRequest{
+		Id: new.Id,
+		UpdateMeta: &apiv2.UpdateMeta{ // TODO: was not wired in?
```

Comment:
Looks like I've forgotten it.

```suggestion
		UpdateMeta: &apiv2.UpdateMeta{
```